client/app/components/_pickupList/pickupList.controller.js

```diff
@@ -0,0 +1,107 @@
+class PickupListController {
+    
+  constructor(Authentication, PickupDate, Store, $filter) {
+    "ngInject";
+    this.reversed = false;
+    this.Authentication = Authentication;
+    this.PickupDate = PickupDate;
+    this.Store = Store;
+    this.userId = -1;
+    this.$filter = $filter;
+        
+        
+    this.pickupList = {
+      showJoined: false,
+      showOpen: true,
+      showFull: false
+    };
+        
+    this.Authentication.update().then((data) => {
+      this.userId = data.id;
+      this.updatePickups();
+    });
+  }
+    
+    /**
+     * adds following infos to a list of pickups:
+     * - isUserMember
+     * - isFull
+     * - store (if showDetail == store)
+     */
+  addPickuplistInfos(pickups) {
+    angular.forEach(pickups, (currentPickup) => {
+      currentPickup.isUserMember = currentPickup.collector_ids.indexOf(this.userId) !== -1;
+      currentPickup.isFull = !(currentPickup.collector_ids.length < currentPickup.max_collectors);
+
+      if (this.showDetail === "store") {
+        currentPickup.store = this.Store.getById(currentPickup.store);
```

Comment:
I think overwrite a value with a promise is bad style. Would it be ok to write this?
```
this.Store.getById(currentPickup.store).then((store) => currentPickup.store = store)
```
does it need a closure?